### PR TITLE
chore(docs): fix inconsistencies in external provisioner docs

### PR DESCRIPTION
**Pull request by johnstcn**
This PR:
- Updates the table in `docs/admin/provisioners.md` to include highlight multi-org changes
- Updates the instructions for the provisionerd helm chart when using provisioner keys

### CODE REVIEW

File: docs/admin/provisioners.md [hunk 1]
```diff
@@ -300,25 +302,7 @@ will use in concert with the Helm chart for deploying the Coder server.
...
-   ```
-
-1. Redeploy Coder with the new `values.yaml` to roll out the PSK. You can omit
-   `--version <your version>` to also upgrade Coder to the latest version.
-
-   ```sh
-   helm upgrade coder coder-v2/coder \
-       --namespace coder \
-       --version <your version> \
-       --values values.yaml
```
Comment (johnstcn) [inline docs/admin/provisioners.md:321]: review: this is no longer required when using provisioner keys, removed.

File: docs/admin/provisioners.md [hunk 2]
```diff
@@ -331,13 +315,17 @@ will use in concert with the Helm chart for deploying the Coder server.
          value: "https://coder.example.com"
      replicaCount: 10
    provisionerDaemon:
+     # NOTE: in older versions of the Helm chart, it is required to set this to an empty string.
+     pskSecretName: ""
```
Comment (johnstcn) [inline docs/admin/provisioners.md:319]: review: PR to address this workaround is here: https://github.com/coder/coder/pull/15417

File: docs/admin/provisioners.md [hunk 3]
```diff
@@ -288,8 +291,7 @@ will use in concert with the Helm chart for deploying the Coder server.
    ```sh
    coder provisioner keys create my-cool-key --org default
    # Optionally, you can specify tags for the provisioner key:
-   # coder provisioner keys create my-cool-key --org default --tags location=auh kind=k8s
```
Comment (johnstcn) [inline docs/admin/provisioners.md:291]: review: `--tags` is incorrect -> `--tag A --tag B`

**Review by bpmct — APPROVED** [review]

File: docs/admin/provisioners.md [hunk 4]
```diff
@@ -331,13 +315,17 @@ will use in concert with the Helm chart for deploying the Coder server.
          value: "https://coder.example.com"
      replicaCount: 10
    provisionerDaemon:
+     # NOTE: in older versions of the Helm chart, it is required to set this to an empty string.
```
Comment (bpmct) [inline docs/admin/provisioners.md:318]:
```suggestion
     # NOTE: in older versions of the Helm chart (2.17.0 and below), it is required to set this to an empty string.
```

**Pull request merged**